pytorch_lightning/accelerators/pipe_accelerator.py

```diff
@@ -0,0 +1,101 @@
+import os
+from typing import List
+
+import fairscale.nn.model_parallel as mpu
+import torch
+import torch.distributed as torch_distrib
+from fairscale.nn import Pipe
+from fairscale.nn.pipe.pipeline import PipelineStyle
+from torch import nn
+from torch.distributed import rpc
+from torch.nn.parallel import DistributedDataParallel
+
+from pytorch_lightning import LightningModule
+from pytorch_lightning import _logger as log
+from pytorch_lightning.accelerators import DDPAccelerator
+from pytorch_lightning.plugins.ddp_plugin import DDPPlugin
+
+
+def get_worker_map():
+    return {rank: f"Test{rank}" for rank in range(torch_distrib.get_world_size())}
+
+
+class LightningPipeModule(nn.Module):
+    def __init__(self, module: nn.Sequential, layer_partitions: List[int], microbatches: int = 8):
+        super().__init__()
+        self.module = module
+        self.layer_partitions = layer_partitions
+        self.microbatches = microbatches
+
+    def init_pipe(self):
+        self.module = Pipe(
+            module=self.module,
+            balance=self.layer_partitions,
+            chunks=self.microbatches,
+            style=PipelineStyle.MultiProcess,
+            input_device=torch.cuda.current_device(),
+            worker_map=get_worker_map(),
+            checkpoint='never',
```

Comment:
What is `checkpoint` never ?

Reply:
The `checkpoint` argument refers to activation checkpointing, i.e. not saving grads on the initial forward pass, and then re-running the forward pass with grads enabled during the backward pass.

Reply:
Will be exposed eventually of course :)